pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -336,7 +336,7 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 		}
 
 		// useCSIPlugin will check both CSIMigration and the plugin specific feature gates
-		if useCSIPlugin(og.volumePluginMgr, volumeToAttach.VolumeSpec) && nu {
+		if useCSIPlugin(og.volumePluginMgr, volumeToAttach.VolumeSpec, &volumeToAttach.NodeName) && nu {
```

Comment:
attach/detach should already be checking this with `nodeUsingCSIPlugin` I think. I want to avoid doing this check twice for controller operations, it may only need to be added for node operations

Reply:
Passing nil in this case - check in nodeUsingCSIPlugin would be effective